api/opds.py

```diff
@@ -592,9 +596,23 @@ def lane_url(self, lane, facets=None):
         return url
 
     def annotate_work_entry(self, work, active_license_pool, edition, identifier, feed, entry):
-        updated = None
-        if isinstance(self.lane, CrawlableCustomListBasedLane) and isinstance(work, BaseMaterializedWork):
-            updated = max(work.last_update_time, work.first_appearance, work.availability_time)
+        updated = work.last_update_time
+        if isinstance(work, WorkSearchResult):
+            # Elasticsearch puts this field in a list, but we've set it up
+            # so there will be at most one value.
+            last_updates = getattr(work._hit, 'last_update', [])
+            if last_updates:
+                # last_update is seconds-since epoch; convert to UTC datetime.
+                search_update = datetime.datetime.utcfromtimestamp(
+                    last_updates[0]
+                )
+                # There's a chance that work.last_updated has been
+                # modified but the change hasn't made it to the search
+                # engine yet.
+                #
+                # TODO: Is it better to use the later value, which is more
+                # accurate, or to ensure that the feed stays sorted?
+                updated = max(updated, search_update)
```

Comment:
I would like your take on this.

Reply:
If you use the search engine value, someone who imports the feed will update this work, and the next time they pull the feed they'll have to update it again even if there are no changes.

If you use the later value, someone who imports the feed may record that timestamp as their last update time, but there may also be new works that have not made it to the search index yet and have timestamps earlier than this one. Next time they pull the feed, they may not reach those works since they'll stop at this later one they've already seen.

The second case seems worse so I would go with the earlier value from the search engine.